### PR TITLE
fix3 - installation instructions

### PR DESCRIPTION
**Pull request by ValeSora**
- Added the `environment.yaml` file for the creation of a `conda` environment.
- Added the `requirements.txt` file for the creation of a `virtualenv` virtual environment.
- Added the `INSTALL.md` file with detailed installation instructions.
- Modified the `setup.py` file to fix the `dask` and `dask.distributed` dependencies.

### CODE REVIEW

**Review by mtiberti — CHANGES_REQUESTED** [review]
there is another general problem: the installation instructions don't include anything about pyinteraph, which is a general requirement. This would be fine, if we weren't instructing the user to install PyInKnife in another virtualenv; this means that they will have two separate virtualenvs if the user follows the instructions for both programs. I think it would be better to instruct the user how to install PyInteraph in the same env. 

As far as the normal virtualenv is concerned, this is not a problem since we can have two requirements.txt (as long as they don't conflict), once for each package.

For conda it would be more complicated since an env can only be defined using a single yaml file (I looked). One way to tackle this would be to add the requirements to PyInteraph to the yaml file here in PyInKnife (and specify this in the docs). This way you could just create the env, then install PyInteraph, then install PyInKnife

File: environment.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    - mock==4.0.3
+    - msgpack==1.0.2
+    - networkx==2.5.1
+    - numpy==1.20.2
+    - packaging==20.9
+    - pandas==1.2.4
+    - partd==1.2.0
+    - Pillow==8.2.0
+    - psutil==5.8.0
+    - pkg-resources==0.0.0
```
Comment (mtiberti) [inline environment.yaml:38]: this is here because of a bug in Debian-based distros and can be removed

**Review by mtiberti — CHANGES_REQUESTED** [review]

File: INSTALL.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+If `git` is not available to you, you can download the repository content as a ZIP file from the PyInKnife2 GitHub repository web page.
+
+#### Step 8 - Install the required Python packages for PyInKnife2
+
+Install all the required Python packages as specified in the `requirements.txt` file in the `PyInKnife2` directory via `pip`:
+
+```shell
+pip install -r requirements.txt
+```
```
[the remark this answers is not in the document]
Comment (ValeSora) [reply INSTALL.md:107]: I encountered compilation issues with the installation of MDAnalysis in a clean environment for 1.0.0 but everything works fine for 1.1.1. I think we should recommend the second one for PyInteraph too.

File: environment.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    - cython==0.29.23
+    - GridDataFormats==0.5.0
+    - gsd==2.4.2
+    - HeapDict==1.0.1
+    - importlib-metadata==4.0.1
+    - iniconfig==1.1.1
+    - joblib==1.0.1
+    - kiwisolver==1.3.1
+    - locket==0.2.1
+    - matplotlib==3.4.1
```
[the remark this answers is not in the document]
Comment (ValeSora) [reply environment.yaml:26]: Ok! Weird, I took it from the packages installed in the clean environment I installed PyInKnife2 in. I'll modify it anyway.

File: environment.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    - MDAnalysis==1.1.1
+    - mmtf-python==1.1.2
+    - mock==4.0.3
+    - msgpack==1.0.2
+    - networkx==2.5.1
+    - numpy==1.20.2
+    - packaging==20.9
+    - pandas==1.2.4
```
[the remark this answers is not in the document]
Comment (ValeSora) [reply environment.yaml:34]: I took it from the pip freeze command :( and it worked when I used the file. But I can modify it to the older version.

File: environment.yaml (hunk 1 above)
Comment (ValeSora) [reply environment.yaml:38]: Ok perfect.

File: environment.yaml [hunk 5]
```diff
@@ -0,0 +1,57 @@
+name: pyinknife2-env
+channels:
+  - conda-forge
+dependencies:
+  pip:
```
[the remark this answers is not in the document]
Comment (ValeSora) [reply environment.yaml:5]: Ok, I'll add the dash! For me it worked also without the dash before `pip`, weird enough (even before the addition of the requirements for PyInteraph2, when I tested it on the server). Should I keep the `channels` part?

File: INSTALL.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+### Installing with `conda`
+
+#### Step 1 - Install `conda`
+
+We have provided an `environment.yml` file that can be used together with `conda` to automatically install an environment contaning all of PyInKnife2's requirement. In order to use it you need to have `conda` installed in your system. Go [here](https://docs.conda.io/en/latest/miniconda.html) for further instructions. Installing `miniconda` rather than full `anaconda` is advised.
+
+Once `conda` is installed on your system, you can use it to create a virtual environment similarly to what you would do using the `virtualenv` package, as previously detailed.
+
+#### Step 2 - Create the `conda` environment
+
```
[the remark this answers is not in the document]
Comment (ValeSora) [reply INSTALL.md:143]: Modified!

File: INSTALL.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+pip install -r requirements.txt
+```
+
+#### Step 6 - Install PyInteraph2
+
+You can now install PyInteraph2:
+
+```shell
+python setup.py install
+```
```
[the remark this answers is not in the document]
Comment (ValeSora) [reply INSTALL.md:86]: Done!

File: INSTALL.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+
+In order to use it you need to have `conda` installed in your system. Go [here](https://docs.conda.io/en/latest/miniconda.html) for further instructions. Installing `miniconda` rather than full `anaconda` is advised.
+
+Once `conda` is installed on your system, you can use it to create a virtual environment similarly to what you would do using the `virtualenv` package, as previously detailed.
+
+#### Step 3 - Create the `conda` environment
+
+Create your `conda` environment starting from the provided environment file:
+
+```shell
```
[the remark this answers is not in the document]
Comment (ValeSora) [reply INSTALL.md:146]: Done!

File: INSTALL.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+
+If `git` is not available to you, you can download the repository content as a ZIP file from the PyInteraph2 GitHub repository web page.
+
+#### Step 6 - Install PyInteraph2
+
+You can now install PyInteraph2:
+
+```shell
+python setup.py install
+```
```
[the remark this answers is not in the document]
Comment (ValeSora) [reply INSTALL.md:185]: Ok, I'll wait for your suggestions :)

File: environment.yaml [hunk 10]
```diff
@@ -0,0 +1,58 @@
+name: pyinknife2-env
+channels:
+  - conda-forge
+dependencies:
```
Comment (mtiberti) [inline environment.yaml:4]: there's a double `- pip:` here. This section should be something like

```
dependencies:
    - python==3.7
    - pip
    - pip:
        - pakage1=...
 ```

File: environment.yaml [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+      - locket==0.2.1
+      - matplotlib==3.3.4
+      - MDAnalysis==1.1.1
+      - mmtf-python==1.1.2
+      - mock==4.0.3
+      - msgpack==1.0.2
+      - networkx==2.5.1
+      - numpy==1.20.2
+      - packaging==20.9
+      - pandas==1.2.4
```
Comment (mtiberti) [inline environment.yaml:36]: doesn't pip crash for you here complaining that the maximum available version is 1.1.5?

**Review by mtiberti — CHANGES_REQUESTED** [review]

File: environment.yaml (hunk 10 above)
Comment (ValeSora) [reply environment.yaml:4]: ok!

File: environment.yaml (hunk 11 above)
Comment (ValeSora) [reply environment.yaml:36]: Now yes. I was using an environment where pandas was somehow already installed (I don't know if it got installed with another package as a dependency). I'll correct it!

**Review by mtiberti — APPROVED** [review]

**Pull request merged**